modules/Core/classes/Core/Notification.php

```diff
@@ -68,7 +70,13 @@ public function __construct(
             $languageCodes = array_column($languageCodes, 'short_code', 'id');
         }
 
-        $this->_recipients = array_map(static function ($recipientId) use ($content, $contentCallback, $skipPurify, $title, $languageCodes) {
+        $notificationPreferences = DB::getInstance()->query(
+            'SELECT `user_id`, `alert`, `email` FROM nl2_users_notification_preferences WHERE `type` = ? AND `user_id` IN (' . implode(',', array_map(static fn ($_) => '?', $recipients)) . ')',
+            array_merge([$this->_type], $recipients)
+        )->results();
+        $notificationPreferences = array_column($notificationPreferences, null, 'user_id');
```

Comment:
Doing this all at once to help remove performance issues with the previous n+1 query